File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.ts
function longestOnes(nums: number[], k: number): number {
    let left = 0, right = 0;
    while (right < nums.length){
        if (!nums[right++]) k--
        if (k < 0)    
            if (!nums[left++]) k++
        
    }
    return right-left
};

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        start = 0
        max_len = 0
        zeros_count = 0
        
        for end in range(len(nums)):
            if nums[end] == 0:
                zeros_count += 1
                
            while zeros_count > k:
                if nums[start] == 0:
                    zeros_count -= 1
                start += 1
            max_len = max(max_len, end - start + 1)
            
        return max_len